src/viam/ur/module/ur_arm.cpp

```diff
@@ -754,40 +787,57 @@ void URArm::move_joint_space_(std::shared_lock<std::shared_mutex> config_rlock,
                 trajex_opts.max_acceleration(i) = current_state_->get_acceleration();
             }
 
-            const auto generation_start = std::chrono::steady_clock::now();
-            auto trajex_path = totg::path::create(trajex_waypoints,
-                                                  totg::path::options{}.set_max_deviation(current_state_->get_path_tolerance_delta_rads()));
-            auto trajex_trajectory = totg::trajectory::create(std::move(trajex_path), std::move(trajex_opts));
-
-            auto sampler = totg::uniform_sampler::quantized_for_trajectory(trajex_trajectory, types::hertz{k_sampling_freq_hz});
-
-            std::vector<trajectory_sample_point> trajex_samples;
-            double previous_time = 0.0;
-            for (const auto& sample : trajex_trajectory.samples(sampler)) {
-                const double current_time = sample.time.count();
-                const float timestep = boost::numeric_cast<float>(current_time - previous_time);
-
-                trajectory_sample_point point;
-                for (size_t i = 0; i < 6; ++i) {
-                    point.p[i] = sample.configuration(i);
-                    point.v[i] = sample.velocity(i);
+            std::vector<trajectory_sample_point> all_trajex_samples;
+
+            for (const auto& segment : segments) {
+                const auto segment_xarray = eigen_waypoints_to_xarray(segment);
+                const totg::waypoint_accumulator trajex_waypoints(segment_xarray);
+
+                auto trajex_path = totg::path::create(
+                    trajex_waypoints, totg::path::options{}.set_max_deviation(current_state_->get_path_tolerance_delta_rads()));
+
+                // Create a copy of trajex_opts for each segment since it's consumed by create()
+                totg::trajectory::options segment_opts = trajex_opts;
+                auto trajex_trajectory = totg::trajectory::create(std::move(trajex_path), std::move(segment_opts));
+
+                auto sampler = totg::uniform_sampler::quantized_for_trajectory(trajex_trajectory, types::hertz{k_sampling_freq_hz});
+
+                double previous_time = 0.0;
+                for (const auto& sample : trajex_trajectory.samples(sampler)) {
```

Comment:
Unlike `sampling_func` below, the `uniform_sampler` is going to give you the t_0 sample back (but please double check me on that). I think this should probably be `for (const auto& sample : trajex_trajectory.samples(sampler) | std::views::drop(1)) {`.